[Yun Ru] iP

yunruu opened the pull request:
# IANA
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

### Iana is a simple task tracker that remembers your tasks for you. It's
* GUI-based
* Easy to learn
* *SUPER* fast to use
* Completely **FREE**

### Try it right **now** by
1. Downloading from [here](https://github.com/yunruu/ip/releases/tag/A-Jar)
2. Double-click it
3. Add your tasks
4. Let it manage your tasks :metal:

### Features:
- [x] Manage (add, delete, mark) tasks
- [ ] Manage deadlines (coming soon)
- [ ] Reminders (coming soon)


### For java programmers, this is the `main` method:
```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

yunruu committed:
First commit; level 1 complete.

yunruu committed:
Level 2 completed; can add and list user input

yunruu committed:
Level 3 completed. Added Task class.

yunruu committed:
Edited mark & unmark functionality to print task.

yunruu committed:
Level 4 complete. Create Todo, Event and Deadline.

yunruu committed:
Complete level 5. Implement DukeException.

yunruu committed:
Complete level 5. Add delete task, more exceptions

yunruu committed:
Edit commit message: level 6 completed.

yunruu committed:
Change UI

yunruu committed:
Add DataLoader to load saved data

yunruu committed:
Change bot name, add save and load data functions

yunruu committed:
Improve error handling

yunruu committed:
Resolve conflict

yunruu committed:
Revert "Change UI"

This reverts commit 76caced81bd55e3eb861e4debb2821ef32ed5b80.

yunruu committed:
Confuse changes

yunruu committed:
Revert "Change bot name, add save and load data functions"

This reverts commit 1e6b918a18ab5bea1a8ad073bf01068cd2bc31be.

yunruu committed:
Reset changes

yunruu committed:
Add load and write data functions

yunruu committed:
Update data storage text

yunruu committed:
Add date time reading function

yunruu committed:
Add date time reading functionality

yunruu committed:
Merge branch-Level-7 with master

yunruu committed:
Merge branch-Level-8 with master

yunruu committed:
Improve OOP and UI; add command classes

yunruu committed:
Package files by classes

yunruu committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

yunruu committed:
Revert "Package files by classes"

This reverts commit ffe1b4b71e641c0631f898b171d5773baf7a6fca.

Dilysss commented on src/main/java/iana/command/Actions.java:
```diff
@@ -0,0 +1,12 @@
+package iana.command;
+
+public enum Actions {
```
Good job for using Enums!

Dilysss commented on src/main/java/iana/command/Command.java:
```diff
@@ -0,0 +1,15 @@
+
+public abstract class Command {
+    
+    public Command() {
+    }
```
Perhaps this lines of code can be omitted

Dilysss commented on src/main/java/iana/command/ListCommand.java:
```diff
@@ -0,0 +1,22 @@
+
+public class ListCommand extends Command {
+    
+    public ListCommand() {
+
+    }
```
Similar to ExitCommand, can remove these lines too

Dilysss commented on src/main/java/iana/tasks/Task.java:
```diff
@@ -0,0 +1,109 @@
+
+import iana.exception.IanaException;
+
+public class Task {
```
Hmm the making sense of the user control could be handled by the Parser class instead

Dilysss commented on src/main/java/iana/main/Iana.java:
```diff
@@ -0,0 +1,54 @@
+import iana.command.Command;
+import iana.exception.IanaException;
+
+public class Iana {
```
Missing javadocs!!

Dilysss reviewed (commented):
I like how you modified your bot to make it more cute and interesting! Overall code is neat and organised too

yunruu committed:
Add packages for different classes. Add Gradle support.

yunruu committed:
Add javadoc for all classes

yunruu committed:
Adhere code to coding standards

yunruu committed:
Add find tasks with keyword function.

yunruu committed:
Merge with A-JavaDoc and A-CodingStandard

yunruu committed:
Merge with Level 9

24Donovan24 commented on src/main/java/iana/command/AddTaskCommand.java:
```diff
@@ -6,13 +6,23 @@
     private String taskInput;
 
+    /**
+     * Constructor for AddTaskCommand.
+     * @param taskInput the full input that describes the task.
```
Do note to leave a spacing between the description and param for JavaDocs! This applies a couple of methods in your other classes as well!

24Donovan24 reviewed (commented):
LGTM

Eclipse-Dominator commented on src/main/Iana.java:
```diff
@@ -0,0 +1,69 @@
+package iana.main;
```
Hello this file's package does not match the current directory.
Perhaps you meant to place this file under /src/main/java/iana/main instead of /src/main/

Eclipse-Dominator commented on src/main/java/iana/main/Storage.java:
```diff
@@ -0,0 +1,90 @@
+                    case "D":
+                    taskTime = taskArray[3];
+                    taskString = String.format("deadline %s /by %s", taskDescription, taskTime);
+                    taskList.add(Task.of(taskString, isCompleted));
```
Hello, you did not catch ArrayIndexOutOfBoundsException. 
It can occur when you have a line like `D | 0 | ` in the `DataStorage.txt`
Even though you specified `input.split(" \\| ", 4);`, when there is not enough seperators, it can still lead to an array that is smaller than the specified amount of parameters.
Since you are trying to catch corrupted lines, perhaps you can add a check to the array length or catch the OutOfBound to account for this.

Eclipse-Dominator commented on src/main/java/iana/main/Storage.java:
```diff
@@ -0,0 +1,90 @@
+                    break;
+
+                    default:
+                    throw new IanaException("File is corrupted");
```
> The switch statement should have the following form: Note there is no indentation for case clauses.

Perhaps you can consider changing the indentation to fit the coding standard.

Eclipse-Dominator commented on src/main/java/iana/exception/IanaException.java:
```diff
@@ -0,0 +1,15 @@
+    public IanaException(String errorMessage) {
+        super(errorMessage);
+    }
+}
```
Maybe you can consider abstracting out the IanaException class to add subclasses for the exception like CorruptLineException etc in order to avoid the use of magic Strings as well as to classify the different types of errors.

Eclipse-Dominator commented on src/main/java/iana/tasks/Task.java:
```diff
@@ -0,0 +1,131 @@
+
+            default:
+            throw new IanaException("Sorry, this is an invalid task type!! D-:");
+        }
```
> The switch statement should have the following form: Note there is no indentation for case clauses.
Perhaps you can update the formatting of the case clauses to fit the coding standard.

Eclipse-Dominator commented on src/main/java/iana/tasks/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+    }
+
+    /**
+     * Add a new task into the task list.
```
> In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.)

Perhaps would you consider changing it to Adds a new task into the task list. Similarly for the `Return xxx` comments below.

Eclipse-Dominator commented on src/main/data/DataStorage.txt:
```diff
@@ -0,0 +1,6 @@
+T | 1 | read book
+D | 0 | return book | June 6th
+E | 1 | project meeting | Aug 6th 2-4pm
+E | 0 | badminton | 10pm
+T | 1 | eat dinner
+D | 0 | eat meal | 12pm
```
Perhaps you can consider adding this file into .gitignore since this is a personalised information for a user.

Eclipse-Dominator commented on build.gradle:
```diff
@@ -0,0 +1,46 @@
+}
+
+application {
+    mainClassName = "seedu.duke.Duke"
```
The class Duke has been removed. Perhaps you should update this reference to iana.main.Iana as well as moving Iana.java to the specified directory.

Eclipse-Dominator reviewed (commented):
Good work on following the java coding standard! There are minor issues and bugs I have listed below.

yunruu committed:
Change implementation of load and store method for Storage class

yunruu committed:
Add GUI

yunruu committed:
Complete implementation of  GUI

yunruu committed:
Improve code quality.

yunruu committed:
Add personality to IANA.

yunruu committed:
Merge pull request #2 from yunruu/branch-A-CodeQuality

Improve code quality.

yunruu committed:
Merge branch 'master' into branch-A-Personality

yunruu committed:
Merge pull request #3 from yunruu/branch-A-Personality

Add personality to IANA.

yunruu committed:
Improve coding standard.

yunruu committed:
Add assertions

yunruu committed:
Implement JUnit tests

yunruu committed:
Update Read.md

yunruu committed:
Update README.md

yunruu committed:
Update README.md

yunruu committed:
Update README.md

yunruu committed:
Update README.md

yunruu committed:
Update README.md

yunruu committed:
Update README.md

yunruu committed:
Implement welcome message

yunruu committed:
Add welcome message on first launch

yunruu committed:
Merge branch 'master' of https://github.com/yunruu/ip

yunruu committed:
Update README.md

yunruu committed:
Update gradle build and files to be ignored

yunruu committed:
Provide in-App guidance to users

yunruu committed:
Merge branch-C-Help to master

yunruu committed:
Add Javadocs

yunruu committed:
Improve GUI

yunruu committed:
Update User Guide with all current features

yunruu committed:
Debug program and update GUI response

yunruu committed:
Merge branch 'master' of https://github.com/yunruu/ip

yunruu committed:
Add product screenshot to docs

yunruu committed:
Add images

yunruu committed:
Update product screenshot

yunruu committed:
Update User Guide

yunruu committed:
Improve error handling

yunruu committed:
Implement CheckStyle support

yunruu committed:
Merge branch 'master' into branch-A-Assertions

yunruu committed:
Merge pull request #4 from yunruu/branch-A-Assertions

Add assertions

yunruu committed:
Update README.md

yunruu committed:
Update bye command to quit app

Bye command: save tasks data and show goodbye message

Bye command does not close application.

A bye command closing the application will enhance user experience
as it is more convenient to save and close app immediately with
the same command.

Let's update the program to check if input is a bye command and
close the application immediately if so.

yunruu committed:
Merge branch 'master' of https://github.com/yunruu/ip

yunruu committed:
Update find command to be case-insensitive

Find command: make task matching case-insensitive

Find command is case sensitive.

A case insensitive find command will be more user-friendly as they
may not remember the case of the task descriptions.

Let's update the search algorithm to be case-insensitive.

yunruu committed:
Add reminder command

There is no reminder command.

Reminder is useful and more user-friendly as users can quickly
see all the tasks that are still incomplete at one go instead of
unnecessary clutter. Completed tasks information may still be
needed at another time so it need not be deleted.

Let's include a reminder command to track all incomplete tasks
quickly and immediately while keeping completed tasks information.

yunruu committed:
Update reminder command

yunruu committed:
Update reminder command and add to help command

yunruu committed:
Merge branch 'master' of https://github.com/yunruu/ip

yunruu committed:
Include reminder command feature

yunruu committed:
Add config tools

yunruu committed:
Update Gradle and Java FX version

yunruu committed:
Improve coding standard
